example/Binary.hs

```diff
@@ -0,0 +1,389 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Binary (
+  Bit (..),
+  LogicBit (..),
+  _LogicO,
+  _LogicI,
+  _LogicO',
+  _LogicI',
+  Binary,
+  zeroo,
+  poso,
+  gtlo,
+  binaryo,
+  addo,
+  subo,
+  lesso,
+  mulo,
+  divo,
+  example,
+) where
+
+import Control.Lens (Prism, from)
+import Control.Lens.TH (makePrisms)
+import Control.Monad (void)
+import Data.Bifunctor (bimap)
+import Data.Function ((&))
+import Data.Tagged (Tagged)
+import GHC.Generics (Generic)
+
+import Core
+import Goal
+import LogicalBase
+import Match
+import TH
+
+data Bit = O | I deriving (Eq, Show, Generic)
+
+makeLogic ''Bit
+deriving instance Show LogicBit
+
+makePrisms ''LogicBit
+
+_LogicO'
+  :: Prism
+      (Tagged (o, i) LogicBit)
+      (Tagged (o', i) LogicBit)
+      (Tagged o ())
+      (Tagged o' ())
+_LogicO' = from _Tagged . _LogicO . _Tagged
+
+_LogicI'
+  :: Prism
+      (Tagged (o, i) LogicBit)
+      (Tagged (o, i') LogicBit)
+      (Tagged i ())
+      (Tagged i' ())
+_LogicI' = from _Tagged . _LogicI . _Tagged
```

Comment:
These are supposed to be derived in the future, right?

Reply:
yes, they should be derived. I'll open an issue for this

Reply:
#7.